src/proto.rs

```diff
@@ -48,7 +48,10 @@ impl<'a> Decoder for ImapCodec {
             IResult::Incomplete(_) => {
                 return Ok(None);
             },
-            IResult::Error(err) => panic!("error {} during parsing of {:?}", err, buf),
+            IResult::Error(err) => {
+                return Err(io::Error::new(io::ErrorKind::Other,
+                                          format!("error {} during parsing of {:?}", err, buf)));
```

Comment:
`"error {}` -> `"{}` (DRY). Also, I prefer to have precise capitalization, punctuation etc. in any text, especially if users may see it. Admittedly, even this wasn’t even done in `std`, which has poorly written error messages like (`os error 14`) all over the place.

Reply:
I can get behind the removal of "error".